### ISSUE TITLE
Implement MonadFix for Handler

### ISSUE THREAD

**Issue by tathougies** [haskell-servant/servant#1480]
`Handler` is just a newtype over `ExceptT IO`. We should be able to implement `MonadFix` for this using standard newtype deriving, but I don't see any instance for this.

**Comment by alpmestan** [comment]
Indeed, we can derive that one. Would you like to submit the patch @tathougies ?
